pkg/core/resources/apis/mesh/proxytemplate_validator.go

```diff
@@ -12,19 +12,18 @@ import (
 	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
 	"github.com/kumahq/kuma/pkg/core/validators"
 	"github.com/kumahq/kuma/pkg/util/envoy"
+	"github.com/kumahq/kuma/pkg/xds/generator"
 )
 
 var availableProfiles map[string]bool
 var availableProfilesMsg string
 
 func init() {
-	profiles := []string{}
 	availableProfiles = map[string]bool{}
-	for _, profile := range AvailableProfiles {
+	for _, profile := range generator.AvailableProfiles {
```

Comment:
This is computed once. Previously it didnt matter because it was computed using a slice that was never modified. But now that the source slice is mutated multiple times Im worried there might be a race. Im not familiar enough with the code base to be sure but I wanted to at least point it out.

Reply:
Good point. I think we can get rid of `availableProfiles` and `availableProfilesMsg`, make `generator.AvailableProfiles` a `map` and have `validateImports` here use `generator.AvailableProfiles`.